docs/source/library-user-guide/upgrading.md

```diff
@@ -122,6 +122,84 @@ See [issue #17713] for more details.
 _execution plan_ of the query. With this release, `DESCRIBE query` now outputs
 the computed _schema_ of the query, consistent with the behavior of `DESCRIBE table_name`.
 
+### Introduction of `TableSchema` and changes to `FileSource::with_schema()` method
```

Comment:
Note that `TableSchema` was not added in this PR but this PR does use it in a way that introduces a notable breaking change, so I think it's an appropriate time to add it to the upgrade guide.